### PR TITLE
udn, ci, e2e: add a non-IC lane

### PR DESCRIPTION
**Pull request by maiqueb**
We had to update the job names so the user segmentation lanes do not clash (the only difference between them is the IC configuration).

#### What this PR does and why is it needed

This PR introduces an e2e test lane for user defined primary networks on **non interconnect deployments**.

#### Which issue(s) this PR fixes

Fixes #

#### Special notes for reviewers

This PR must go after the PR that fixes the e2e tests for L2 on non-IC deployments: #4517 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by maiqueb** [comment]
This push should address the UDN L2 east / west test . 

We should then open up an issue about the isolation failed tests on non-IC, and merge this PR.

**Comment by maiqueb** [comment]
Seems we're missing the NAD factory on the nodes on non-IC scenarios ... 

I'll try to figure out what's missing here ....

**Comment by maiqueb** [comment]
> Seems we're missing the NAD factory on the nodes on non-IC scenarios ...
> 
> I'll try to figure out what's missing here ....

nil pointer being thrown [here](https://github.com/ovn-org/ovn-kubernetes/blob/cea122aa5e3b52a8efa458dc1b6e995de857ef3b/go-controller/pkg/factory/factory.go#L1472): 
```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x197709c]

goroutine 86 [running]:
github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory.(*WatchFactory).NADInformer(0xc000000000?)
	/home/mduarted/github/ovn-kubernetes/go-controller/pkg/factory/factory.go:1472 +0x1c
github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni.NewCNIServer({0x280b2c0, 0xc000549880}, {0x28101c0?, 0xc000582ea0})
	/home/mduarted/github/ovn-kubernetes/go-controller/pkg/cni/cniserver.go:75 +0x330
github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node.(*DefaultNodeNetworkController).Start(0xc0005481c0, {0x27ee980, 0xc00003c820})
	/home/mduarted/github/ovn-kubernetes/go-controller/pkg/node/default_node_network_controller.go:794 +0x9f2
github.com/ovn-org/ovn-kubernetes/go-controller/pkg/network-controller-manager.(*nodeNetworkControllerManager).Start(0xc0000fe690, {0x27ee980, 0xc00003c820})
	/home/mduarted/github/ovn-kubernetes/go-controller/pkg/network-controller-manager/node_network_controller_manager.go:131 +0x322
main.runOvnKube.func4()
	/home/mduarted/github/ovn-kubernetes/go-controller/cmd/ovnkube/ovnkube.go:542 +0x258
created by main.runOvnKube in goroutine 1
	/home/mduarted/github/ovn-kubernetes/go-controller/cmd/ovnkube/ovnkube.go:520 +0x625
```

**Comment by maiqueb** [comment]
After allowing the node to start the NAD factory, in [this commit](https://github.com/ovn-org/ovn-kubernetes/pull/4518/commits/09b512fc80215ca4e0732978d0186d27ccfc51d2), I get the following RBAC error:
```
versions/factory.go:117: Failed to watch *v1.NetworkAttachmentDefinition: failed to list *v1.NetworkAttachmentDefinition: network-attachment-definitions.k8s.cni.cncf.io is forbidden: User "system:ovn-node:ovn-worker2" cannot list resource "network-attachment-definitions" in API group "k8s.cni.cncf.io" at the cluster scope
```

...

**Comment by maiqueb** [comment]
The failing tests are tracked in #4528.

File: go-controller/pkg/factory/factory.go [hunk 1]
```diff
@@ -666,7 +666,7 @@ func NewNodeWatchFactory(ovnClientset *util.OVNNodeClientset, nodeName string) (
 	}
 
 	// need to configure OVS interfaces for Pods on secondary networks in the DPU mode
-	if config.OVNKubernetesFeature.EnableMultiNetwork && config.OvnKubeNode.Mode == types.NodeModeDPU {
+	if config.OVNKubernetesFeature.EnableMultiNetwork {
```
Comment (tssurya) [inline go-controller/pkg/factory/factory.go:669]: so for secondary networks we don't need this cause we don't watch for NADs in node watch factory huh?

So its really:
(config.OVNKubernetesFeature.EnableMultiNetwork && config.OvnKubeNode.Mode == types.NodeModeDPU) || util.IsNetSegEnabled() && !Interconnect?

File: test/e2e/network_segmentation.go [hunk 2]
```diff
@@ -150,6 +151,13 @@ var _ = Describe("Network Segmentation", func() {
 				netConfigParams networkAttachmentConfigParams,
 				udnPodConfig podConfiguration,
 			) {
+				if !isInterconnectEnabled() {
+					const upstreamIssue = "https://github.com/ovn-org/ovn-kubernetes/issues/4528"
+					e2eskipper.Skipf(
+						"These tests are known to fail on non-IC deployments. Upstream issue: %s", upstreamIssue,
```
Comment (tssurya) [inline test/e2e/network_segmentation.go:157]: nice!

File: go-controller/pkg/factory/factory.go (hunk 1 above)
Comment (maiqueb) [reply go-controller/pkg/factory/factory.go:669]: yeah I guess. 

Not easy to reason about that or explain why though ...

Comment (tssurya) [reply go-controller/pkg/factory/factory.go:669]: nit: adding a comment to that effect would be good, but I can trace it back

File: dist/templates/rbac-ovnkube-node.yaml.j2 [hunk 3]
```diff
@@ -139,6 +139,10 @@ rules:
       resources:
           - endpointslices
       verbs: [ "get", "list", "watch" ]
+    - apiGroups: ["k8s.cni.cncf.io"]
```
Comment (tssurya) [inline dist/templates/rbac-ovnkube-node.yaml.j2:142]: self-note: needed because we run node separately on nonIC

**Review by tssurya — APPROVED** [review]

**Comment by maiqueb** [comment]
Failed upstream on the `control-plane, noHA, shared, ipv6, noSnatGW, 2br, ic-single-node-zones` lane:
```
  [FAIL] e2e egress firewall policy validation with external containers [It] Should validate the egress firewall policy functionality for allowed IP
```
@tssurya is this a known error ?

**Comment by flavio-fernandes** [comment]
hi @maiqueb and @tssurya 

Is it possible that this PR broke `kind-helm.sh -mne`?
I will dig into it a little more but it seems that while using latest master build, I get the following crash:

```
I0716 18:28:36.852466    1484 reflector.go:325] Listing and watching *v1.NetworkAttachmentDefinition from github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/informers/externalversions/factory.go:117
W0716 18:28:36.853209    1484 reflector.go:539] github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/informers/externalversions/factory.go:117: failed to list *v1.NetworkAttachmentDefinition: network-attachment-definitions.k8s.cni.cncf.io is forbidden: User "system:ovn-node:ovn-worker2" cannot list resource "network-attachment-definitions" in API group "k8s.cni.cncf.io" at the cluster scope
```

```
❯ git bisect bad
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[11c8637b43dbdc2b9d96e462ea04bb3c332f4c23] udn, ci, e2e: add a non-IC lane
❯ git bisect good
d1a012575dd524d1aee675770ba9ed2ab8b0a204 is the first bad commit
commit d1a012575dd524d1aee675770ba9ed2ab8b0a204
Author: Miguel Duarte Barroso <mdbarroso@redhat.com>
Date:   Mon Jul 15 19:53:21 2024 +0200

    udn, non-IC: start NAD listener

    For the UDN feature, we now need to read the NADs from the node (so the
    CNI knows it needs to wait for the annotations of the user defined
    primary network).

    Thus, we need to start the NAD factory also when the user defined
    primary networks feature is enabled.

    Signed-off-by: Miguel Duarte Barroso <mdbarroso@redhat.com>

 go-controller/pkg/factory/factory.go | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)
```

We do not have a lane in github with `kind-helm -mne`, so I do not think you did anything wrong.

**Comment by maiqueb** [comment]
Yeah ... Sorry about that. 

Do you know which role needs to be patched ? FWIW, what's missing is patching it up as was done in [this commit](https://github.com/ovn-org/ovn-kubernetes/pull/4518/commits/9e2aa33cf0bce0994ae96a1be982528ee16471a2).

If you let me know, I'll push a PR to fix it. If you want to do it yourself, add me as reviewer.

Thanks.

**Comment by flavio-fernandes** [comment]
> Yeah ... Sorry about that.
> 
> Do you know which role needs to be patched ? FWIW, what's missing is patching it up as was done in [this commit](https://github.com/ovn-org/ovn-kubernetes/pull/4518/commits/9e2aa33cf0bce0994ae96a1be982528ee16471a2).
> 
> If you let me know, I'll push a PR to fix it. If you want to do it yourself, add me as reviewer.
> 
> Thanks.

no worries. PTAL:  https://github.com/ovn-org/ovn-kubernetes/pull/4535

**Comment by tssurya** [comment]
Yeah sorry @flavio-fernandes ... It's kinda of getting really hard to keep track of all combinations..we support. Thanks for catching this.

**Pull request merged**